cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -51,6 +54,46 @@ func CreateOrUpdateConfigMap(client clientset.Interface, cm *v1.ConfigMap) error
 	return nil
 }
 
+// CreateOrMutateConfigMap tries to create the ConfigMap provided as cm. If the resource exists already, the latest version will be fetched from
+// the cluster and cmmutator callback will be called on it, then an Update of the mutated ConfigMap will be performed. This function is resilient
+// to conflicts, and a retry will be issued if the configmap was modified on the server between the refresh and the update (while the mutation was
+// taking place)
+func CreateOrMutateConfigMap(client clientset.Interface, cm *v1.ConfigMap, cmmutator ConfigMapMutator) error {
+	if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Create(cm); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "unable to create configmap")
+		}
+		return MutateConfigMap(client, metav1.ObjectMeta{Namespace: cm.ObjectMeta.Namespace, Name: cm.ObjectMeta.Name}, cmmutator)
+	}
+	return nil
+}
+
+// MutateConfigMap takes a ConfigMap Object Meta (namespace and name), retrieves the resource from the server and tries to mutate it
+// by calling to the cmmutator callback, then an Update of the mutated ConfigMap will be performed. This function is resilient
+// to conflicts, and a retry will be issued if the configmap was modified on the server between the refresh and the update (while the mutation was
+// taking place)
+func MutateConfigMap(client clientset.Interface, cmmeta metav1.ObjectMeta, cmmutator ConfigMapMutator) error {
```

Comment:
`cmmeta` and `cmmutator` are not very readable to me. `meta` and `mutator` seem good enough. Since the func is called `MutateConfigMap` the context is clear to that these are the config map meta and config map mutator.